src/gui/src/viewer/zoom-window.cpp

```diff
@@ -1072,7 +1111,50 @@ void ZoomWindow::toggleSlideshow()
 	}
 }
 
-
+void ZoomWindow::resizeButtons()
```

Comment:
This is clearly a work-in-progress function, but doing such calculations, especially stuff like `findChild()` shouldn't be necessary if the layouts are set correctly.

Aren't those already using [QHBoxLayout|https://doc.qt.io/qt-5/qhboxlayout.html] and stretch factors?

Reply:
Yes, I think there are a few ways of handling that. It would be easy to hard-code a reference to any of the existing action buttons but I wanted to allow for buttons to be added and removed without maintenance. I tried using the stretch factors but couldn't get it to work because the navigation buttons aren't in the actionButtons layout (?). Another option would be to reference a set value on the layout, like min/max size depending on the size policy. I can try to look into that if you like this U.I. change in general. It might even be possible to achieve the same result with a grid layout like you had before.

Reply:
If you want it to work by removing any button, I think it should work with 3 horizontal layouts + 1 vertical layout set directly in the UI file, somehow like this:
![image](https://user-images.githubusercontent.com/882719/111053447-1f338c00-8464-11eb-8ef8-5b2432292f5b.png)

Where  red is a horizontal layout and green is a vertical layout.
* A 6-item "parent" horizontal layout, with a stretch factor of "0,1,3,0,1,0" or something (haven't played with those for a while)
* A vertical layout with 2 items, the "normal" buttons and favorites buttons
* Two horizontal layouts for the buttons

With this it should stretch properly automatically. Otherwise it can stay like this and I'll have a look when you're done if I can manage something 😄